### DISCUSSION TITLE
remoteAdvanced3 not solvable?

### DISCUSSION THREAD

**Discussion by bebehei** [pcottle/learnGitBranching#1166] [General]
In the past, I never played the game, but always remembered the game as a possible course for beginners to learn by visual.

Years later (today), I recommended this to my colleagues and took the opportunity to walk through the game. Most of it was just comparing the output of `show goal` and the actual worksheet.

Now I'm hanging at the `remoteAdvanced3`. The worksheet and the goal look all the same, but the level does not complete. What can I do?

Executed the following steps for reproducibility:

```
git checkout -b side
git commit
git pull --rebase origin main
git push origin side:main
```

![image](https://github.com/user-attachments/assets/54f14f97-e6ab-45de-8001-b7e9d300ecfa)

**Comment by bebehei** [comment]
Ok, after checking out the sources, I found: https://github.com/pcottle/learnGitBranching/blob/main/src/levels/remote/tracking.js

Copying the proposed solution will yield a success. (I do not spoiler it here)

And after inspecting the `goalTreeString` in the js file, I found a `remoteTrackingBranch` attribute for the `side` branch set to `o/main`.

I suspect this is the issue, since my "solution" does not set tracking info.

---

I'd appreciate as the remote tracking branches are compared, too, the remote tracking branches should be visualized, so that you see them with `show goal` too.

**Reply by pcottle** [reply]
You can also just type in `show solution` for the solution to be demonstrated for you :)

**Reply by bebehei** [reply]
Well, I did not know about `show solution` yet. There's a hint in the text during `show solution`'s execution, that the remote tracking branch was set up.

Still goal and worksheet are visually equal, but in fact differ internally. Personally, I find this unfortunate. The game's amazingness is **visualizing** how git works.
